FrontEnd/src/components/SearchPage/search_field/SearchResults.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from "react";
+import {MainCompanies} from './companies/Companies';
+import './Text.css'
+
+const ITEMS_PER_PAGE = 6;
+
+const SearchResults = ({ results, searchTerm }) => {
+    let error = null;
+
+    if (results && results.error) {
+        error = results.error;
+    }
+
+    const [currentPage, setCurrentPage] = useState(1);
+    const totalItems = results.length;
+    const totalPages = Math.ceil(totalItems / ITEMS_PER_PAGE);
+    
+    const startIndex = (currentPage - 1) * ITEMS_PER_PAGE;
+    const endIndex = startIndex + ITEMS_PER_PAGE;
+    const displayedResults = results.slice(startIndex, endIndex);
+
+    const handlePageChange = (newPage) => {
+        setCurrentPage(newPage);
+    };
+
+    return (
+        <div>
+            {!error && (
+                <>
+                    <div className="new-companies-block">
+                        <div className="row">
```

Comment:
Semantically, its better for tables to use appropriate elements such as `table, tr, th, td ...`